app/controllers/vacancies_controller.rb

```diff
@@ -10,13 +12,9 @@ def new
 
   def create
     @vacancy = Vacancy.new(vacancy_params)
-
-    if @vacancy.save
-      flash[:success] = t(".success")
-      redirect_to @vacancy
-    else
-      render :new
-    end
+    Vacancy::Publish.new(@vacancy).publish
```

Comment:
It makes sense.. I like to inject the object as dependency when I have more methods that needs to use it. But in this case, it makes sense.
What do you think @fpgentil ?